aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -248,11 +258,18 @@ module AptosFramework::Stake {
         // update validator info (so network address/public key change takes effect)
         let i = 0;
         let len = Vector::length(&validator_set.active_validators);
+        let active_validators = Vector::empty();
         while (i < len) {
             let old_validator_info = Vector::borrow_mut(&mut validator_set.active_validators, i);
-            *old_validator_info = generate_validator_info(old_validator_info.addr);
+            let new_validator_info = generate_validator_info(old_validator_info.addr);
+            if (new_validator_info.voting_power >= validator_set.minimum_stake &&
+                new_validator_info.voting_power <= validator_set.maximum_stake
```

Comment:
When will this happen? We already check before adding more stake to the validator that we are not exceeding the max stake limit in `delegate_stake` function?

Reply:
this is on each epoch, we may adjust the maximum_stake requirement though, so I thought it's nice to maintain the invariant